### PR TITLE
UI: FIX #3415 Tweak Manage Gang button visibility

### LINKED ISSUE #3415: Manage Gang button improvements.
Once the player has SF2 the Manage Gang button should always be visible but disabled if they don't have the karma req.

### PR DESCRIPTION
**Pull request by borisflagell**
fixes #3415
image

Testing done :
Ensuring the button stay hidden in the following cases :
- Player not having source-file 2.
- Player not being inside bitnode 2.
- Faction not being Gang-capable.
- Faction being Gang-capable, but player being already  affiliated with another faction.

Ensuring the button is visible but cannot be clicked :
- When the player has the source-file 2, has yet to create a gang, but lack the required karma. 

Ensuring the button is visble and interactable :
- When the player is in bitnode 2.
- When the player has the source-file 2, has yet to create a gang, and has the required karma.
- When the player has created a gang.

### CODE REVIEW

**Review by nickofolas — APPROVED** [review]

File: src/PersonObjects/Player/PlayerObjectGangMethods.ts [hunk 1]
```diff
@@ -17,6 +17,13 @@ export function canAccessGang(this: IPlayer): boolean {
   return this.karma <= GangKarmaRequirement;
 }
 
+export function isAwareOfGang(this: IPlayer): boolean {
+  if (this.bitNodeN === 2 || this.sourceFileLvl(2) >= 1) {
+    return true;
+  }
+  return false;
```
Comment (ApamNapat) [inline src/PersonObjects/Player/PlayerObjectGangMethods.ts:24]:
```suggestion
  return this.bitNodeN === 2 || this.sourceFileLvl(2) >= 1;
```

**Review by ApamNapat — CHANGES_REQUESTED** [review]

File: src/Faction/ui/FactionRoot.tsx [hunk 2]
```diff
@@ -121,9 +125,23 @@ function MainPage({ faction, rerender, onAugmentations }: IMainProps): React.Rea
...
       </Typography>
       <Info faction={faction} factionInfo={factionInfo} />
-      {canAccessGang && (
+      {isManageGangVisible && (
         <>
-          <Option buttonText={"Manage Gang"} infoText={gangInfo} onClick={manageGang} />
+          <Box>
+            <Paper sx={{ my: 1, p: 1 }}>
+              <Tooltip
+                title={!isManageGangClickable ? <Typography>Unlocked when reaching -54000 karma</Typography> : ""}
```
Comment (phyzical) [inline src/Faction/ui/FactionRoot.tsx:133]: do we have a constant to back this on? `-54000` just incase the number was changed someday in the future

**Review by phyzical — APPROVED** [review]

File: src/Faction/ui/FactionRoot.tsx (hunk 2 above)
Comment (borisflagell) [reply src/Faction/ui/FactionRoot.tsx:133]: Not at that point, it's currently a const in PlayerObjectGangMethod.ts. Declared right before canAccessGang function's definition.

Comment (borisflagell) [reply src/Faction/ui/FactionRoot.tsx]: I pushed a commit to streamline that issue.

**Pull request merged**